src/model/scripts/lighting_training/data_preprocessing.py

```diff
@@ -0,0 +1,69 @@
+import argparse
+from openthaigpt_pretraining_model.lightning.utils import (
+    TokenizedDataset,
+)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--mode",
+        type=str,
+        default="train",
+        help="train | val",
+    )
+    parser.add_argument(
+        "--model",
+        type=str,
+        default="decapoda-research/llama-7b-hf",
+        help="train | val",
+    )
+    parser.add_argument(
+        "--max_tokens",
+        type=int,
+        default=2048,
+    )
+    parser.add_argument(
+        "--save_path",
+        type=str,
+        default="./tokendata",
+    )
+    parser.add_argument(
+        "--chunk_size",
+        type=int,
+        default=1024 * 1024,
+    )
+    parser.add_argument(
+        "--batch_size",
+        type=int,
+        default=25000,
+    )
+    parser.add_argument(
+        "--num_proc",
+        type=int,
+        default=128,
+    )
+    parser.add_argument(
```

Comment:
Our input dataset may not be huggingface dataset but multiple jsonl files. How can we handle this
Options I can think of
1. Create simple script to convert jsonl to huggingface dataset and also create simple script to merge multiple hf dataset into one hf dataset. Then we use one hf dataset into this tokenization pipeline.
2. Modify this pipeline to also support multiple jsonl and converted into one hf dataset after tokenized

Reply:
done